src/main/java/net/glowstone/block/itemtype/ItemBow.java

```diff
@@ -0,0 +1,116 @@
+package net.glowstone.block.itemtype;
+
+import java.util.Optional;
+import java.util.concurrent.ThreadLocalRandom;
+import net.glowstone.entity.GlowPlayer;
+import net.glowstone.entity.projectile.GlowTippedArrow;
+import net.glowstone.inventory.GlowInventorySlot;
+import net.glowstone.util.InventoryUtil;
+import org.bukkit.GameMode;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Arrow;
+import org.bukkit.entity.SpectralArrow;
+import org.bukkit.entity.TippedArrow;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.PotionMeta;
+
+public class ItemBow extends ItemTimedUsage {
+    private static final long TICKS_TO_FULLY_CHARGE = 20;
+    private static final double MAX_BASE_DAMAGE = 9;
+    private static final double MAX_SPEED = 53;
+
+    @Override
+    public void startUse(GlowPlayer player, ItemStack item) {
+        if (findArrow(player).isPresent()) {
```

Comment:
In vanilla, you can shoot a bow even if you don't have arrows in your inventory. By default, it will shoot plain arrows, and if there are tipped arrows, it will use those instead (without consumption).

Here, the startUse would do nothing if the player is in creative and has no arrows, thus not shooting anything in the end.

Reply:
Fixed: now shoots a plain arrow if you're in Creative and have no arrows.